src/server/search/aggregator.h

```diff
@@ -19,10 +20,16 @@ namespace dfly::aggregate {
 using Value = ::dfly::search::SortableValue;
 using DocValues = absl::flat_hash_map<std::string, Value>;  // documents sent through the pipeline
 
-// TODO: Replace DocValues with compact linear search map instead of hash map
+struct PipelineResult {
+  // Values to be passed to the next step
+  // TODO: Replace DocValues with compact linear search map instead of hash map
+  std::vector<DocValues> values;
 
-using PipelineResult = io::Result<std::vector<DocValues>, facade::ErrorReply>;
-using PipelineStep = std::function<PipelineResult(std::vector<DocValues>)>;  // Group, Sort, etc.
+  // Fields from values to be printed
+  absl::flat_hash_set<std::string> fields_to_print;
```

Comment:
why flat_hash_set

Reply:
This container mainly holds a small number of elements, which makes `flat_hash_set` a good fit here. Do you have any other alternatives?

Reply:
why not simple vector or even InlinedVector

Reply:
In the second PR, we are adding multiple fields to the `fields_to_print` in the SORTBY option, which may already exist there